include/boost/beast/core/async_base.hpp

```diff
@@ -249,6 +249,10 @@ class async_base
     /// Move Constructor
     async_base(async_base&& other) = default;
 
+    virtual ~async_base() = default;
```

Comment:
`virtual` is wrong here, because this class has no `virtual` member functions. Considering the fact that the move constructor is only for exposition in documentation, might as well apply the rule of 0 and wrap it into an ifdef:

```
#if BOOST_BEAST_DOXYGEN
    /// Move Constructor
    async_base(async_base&& other) = default;
#endif
```
This should silence the warning in regular builds.

Reply:
isn't `before_invoke_hook` virtual? If I use your suggestion, I still get the warning for `async_base`.

Reply:
Hmm, I missed that. The destructor should be non-virtual(since this class isn't supposed to be used like a typical polymorphic base class) and `protected`. copy/move operations should be `default`ed.

Reply:
Yeah but making the destructor virtual has almost no cost, and satisfies the pedants.

Reply:
I think I only really need `suite.hpp` changes in order to be able to use the unit_test stuff in our project, so I'd be happy to omit or modify the `async_base` however you see fit. Thanks.

Reply:
I think we should keep the destructor virtual. The class already has a virtual member, and making the destructor virtual will silence possible warnings. The cost is so miniscule as to be virtually nil.